File: _2_a_data_structure_in_cpp/_1_arrays_and_strings/_6_sliding_window/_1_fixed_size_with_simple_calculation/_1_maximum_average_subarray.cpp
// https://leetcode.com/problems/maximum-average-subarray-i/

class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int i=0,j=0,n=nums.size();

       double sum=0;
       double maxi = INT_MIN;
        while(j<n) {
         sum+=nums[j];

         if(j-i+1==k) {
             double avg = sum/k;
             maxi = max(maxi, avg);
             sum-=nums[i];
             i++;
         }

         j++;
        }

        return maxi;
    }
};

File: _2_data_structure_new/_1_arrays_and_strings/_6_sliding_window/_1_fixed_size_with_simple_calculation/_1_maximum_average_subarray/_1_maximum_average_subarray.java
// https://leetcode.com/problems/maximum-average-subarray-i/

import java.util.*;

class Solution {
    public double findMaxAverage(ArrayList<Integer> nums, int k) {
        int i = 0, j = 0, n = nums.size();

        double sum = 0;
        double maxi = Double.NEGATIVE_INFINITY;

        while (j < n) {
            sum += nums.get(j);

            if (j - i + 1 == k) {
                double avg = sum / k;
                maxi = Math.max(maxi, avg);
                sum -= nums.get(i);
                i++;
            }

            j++;
        }

        return maxi;
    }
}
